org.eclipse.lyo.oslc4j.codegenerator/src/org/eclipse/lyo/oslc4j/codegenerator/files/generateServiceProviderCatalogSingleton.mtl

```diff
@@ -165,37 +161,17 @@ public class [javaClassNameForSingleton(aServiceProviderCatalog) /]
     }
 
     [for (aServiceProvider: ServiceProvider | aServiceProviderCatalog.serviceProviders)]
-
-    private static URI construct[aServiceProvider.javaName(true)/]URI([aServiceProvider.instanceMethodSignature() /])
-    {
-        String basePath = OSLC4JUtils.getServletURI();
-        Map<String, Object> pathParameters = new HashMap<String, Object>();
-        [for (instanceCompositeID: String | aServiceProvider.instanceCompositeID()) separator(lineSeparator())]
-        pathParameters.put("[instanceCompositeID /]", [instanceCompositeID /]);
-        [/for]
-        String instanceURI = "[aServiceProvider.instanceURI() /]";
-
-        final UriBuilder builder = UriBuilder.fromUri(basePath);
-        return builder.path(instanceURI).buildFromMap(pathParameters);
+    public static boolean contains[aServiceProvider.javaName(true)/](final [javaClassName(aServiceProvider) /] serviceProviderInfo) {
```

Comment:
contains method should either check by the `identifier` or by a `ServiceProvider` object, not the SPInfo

Reply:
The method initServiceProviders() deals with SPInfo, so I thought it easier to simply send those SPInfo instances around. Plus, there is no SP instance yet created when this containsSP method is called.
But sure, we can **also** have a contains method that takes an identifier. But at the moment that method is not used. Are you planning to call these methods from outside the SPCSingleton?

Reply:
I deal with the SPInfo only once. Then I get the created SP and from that moment on I discard the SPInfo. SPs go into the triplestore and back. So, SPInfo methods are really not very convenient for me. I would have to take an SP, create an SPInfo from it to check if the SP is registered... duh.

Reply:
I see. Let me see what I can do. on it.